hub/core/link_creds.py

```diff
@@ -132,11 +132,11 @@ def populate_creds(self, creds_key: str, creds):
         expires_in_to_expires_at(creds)
         self.creds_dict[creds_key] = creds
 
-    def add_to_used_creds(self, creds_key: str):
-        if creds_key not in self.used_creds_keys:
-            self.used_creds_keys.add(creds_key)
-            return True
-        return False
+    def add_to_used_creds(self, creds_key: Optional[str]):
```

Comment:
Should we allow this function call with `creds_key: None`?

Reply:
I don't see the harm, creds_key, in general, can be None when not specified. It will always return False in this and not get added. Alternative is to check for None externally whereever add_to_used_creds is called and not call it, which could be more error prone as we'll have to do it in multiple places.